native/src/components/Header.tsx

```diff
@@ -213,18 +214,35 @@ const Header = ({
     if (currentRouteIsPoi && notFromDeepLink) {
       const poisRouteParams = route.params as RoutesParamsType[PoisRouteType]
       if (poisRouteParams.slug || poisRouteParams.multipoi !== undefined) {
-        return t('locations')
+        return { text: t('locations'), language: undefined } // system language
       }
     }
 
-    const previousRouteTitle = (previousParams as { title?: string } | undefined)?.title
-    return previousRouteTitle ?? t(previousRoute.name === 'categories' ? 'localInformation' : previousRoute.name)
+    const previousRouteTitle = (previousParams as { path?: string } | undefined)?.path
+
+    if (previousRouteTitle) {
+      return { text: previousRouteTitle, language: languageCode }
+    }
+
+    if (previousRoute.name === CATEGORIES_ROUTE) {
+      return {
+        text: t('localInformation'),
+        language: previousRouteTitle === undefined ? config.sourceLanguage : languageCode,
+      }
+    }
+
+    return { text: previousRoute.name, language: undefined } // system language
```

Comment:
:wrench: Feel free to fix this CodeScene issue, if you are motivated.

Reply:
I hadn't even noticed it, thanks! Looking into it also meant finally looking into why my emulator hasn't been updating in forever so that I could install the CodeScene extension, so thank you for that too!